modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java

```diff
@@ -141,23 +145,36 @@ public int getSegmentsEntryClassPKsCount(long segmentsEntryId)
 		List<SegmentsEntry> segmentsEntries =
 			segmentsEntryLocalService.getSegmentsEntries(
 				groupId, true, getSource(), className, QueryUtil.ALL_POS,
-				QueryUtil.ALL_POS, null);
+				QueryUtil.ALL_POS,
```

Comment:
Would be awesome if we could stop sorting this query.

Reply:
what do you mean? passing a null value as OrderByComparator?